src/main/java/com/linkedin/kmf/apps/MirrorPipelineMonitor.java

```diff
@@ -0,0 +1,448 @@
+ * broker in the pipeline.
+ *
+ * MirrorPipelineMonitor app exports these metrics via JMX. It also periodically report metrics if DEBUG level logging
+ * is enabled. This information can be used by other application to trigger alert when kafka brokers fail. It also
+ * allows users to track pipeline performance, e.g. latency and throughput.
+ */
+
+public class MirrorPipelineMonitor implements App {
+  private static final Logger LOG = LoggerFactory.getLogger(MirrorPipelineMonitor.class);
+
+  private final ProduceService _produceService;
+  private final ConsumeService _consumeService;
+  private final List<TopicManagementService> _topicManagementServices;
+  private final String _name;
+
+  public MirrorPipelineMonitor(Map<String, Object> props, String name) throws Exception {
+    _name = name;
+    _produceService = new ProduceService(createProduceServiceConfig(props), name);
+    _consumeService = new ConsumeService(createConsumeServiceConfig(props), name);
+    _topicManagementServices = new ArrayList<>();
+    createTopicManagementServices(props, name);
+  }
+
+  private void createTopicManagementServices(Map<String, Object> props, String name) {
+    Map<String, Object> topicManagementConfig = createTopicManagementServiceConfig(props);
+    Double partitionsToBrokerRatio;
+    if (topicManagementConfig.containsKey(TopicManagementServiceConfig.PARTITIONS_TO_BROKER_RATIO_CONFIG)) {
+      partitionsToBrokerRatio = getDouble(topicManagementConfig, TopicManagementServiceConfig.PARTITIONS_TO_BROKER_RATIO_CONFIG);
+    } else {
+      partitionsToBrokerRatio = TopicManagementServiceConfig.PARTITIONS_TO_BROKER_RATIO_DEFAULT;
+    }
+
+    List<String> zookeeperUrlsList = getList(topicManagementConfig, MirrorPipelineMonitorConfig.ZOOKEEPER_CONNECT_LIST_CONFIG);
+    if (zookeeperUrlsList == null) {
+      throw new RuntimeException(_name + ": Must specify at least one zookeeper connection.");
+    }
+
+    // Mirrored topics must have the same number of partitions but may have a different
+    // number of brokers. Separate partitionToBrokerRatios must be maintained to achieve this requirement.
+    Map<String, Integer> zookeeperBrokerCount = new HashMap<>();
+    int maxNumPartitions = 0;
+    for (String zookeeper: zookeeperUrlsList) {
+      if (zookeeperBrokerCount.containsKey(zookeeper)) {
+        throw new RuntimeException(_name + ": Duplicate zookeeper connections specified for topic management.");
+      }
+
+      int brokerCount = com.linkedin.kmf.common.Utils.getBrokerCount(zookeeper);
+      int numPartitions = (int) Math.ceil(partitionsToBrokerRatio * brokerCount);
+      maxNumPartitions = Math.max(maxNumPartitions, numPartitions);
+      zookeeperBrokerCount.put(zookeeper, brokerCount);
+    }
+
+    for (String zookeeper: zookeeperUrlsList) {
+      topicManagementConfig.put(ConsumeServiceConfig.ZOOKEEPER_CONNECT_CONFIG, zookeeper);
+      double partitionToBrokerRatio = maxNumPartitions / zookeeperBrokerCount.get(zookeeper);
```

Comment:
When brokers are added to one cluster or another the ratio might need to change.  A better way to do this might be to supply some kind of additional object to the topic management service that is the policy that describes the desired number of topics.  That would be a different patch.